onnxruntime/core/optimizer/layout_transformation/layout_transformation.cc

```diff
@@ -177,7 +177,11 @@ Status TransformLayoutForEP(Graph& graph, bool& modified, const IExecutionProvid
         for (size_t i = 2; i < node->Inputs().size(); i++) {
           auto constant = api_graph->GetConstant(node->Inputs()[i]);
           if (constant != nullptr && constant->Data().size() > 0) {
-            input_perms.push_back(&input_perm);
+            // Starting from opset version 18, 'scales' and 'sizes' can be 2D tensors.
+            // However, our current implementation only supports the transposition of 4D tensors.
```

Comment:
This layout transformer class appears narrowly limited to only 4D rather than generic, based on comments elsewhere (mentioning NCHW, but not NCW or NCDHW), whereas Resample itself is a very generic like reduction that just takes axes. So yeah, until this class is more properly generic, this check makes sense.